// @wq/map/components/GeoTools.js
import React from 'react';
import { useComponents, useInputComponents } from '@wq/react';
import { useMinWidth } from '@wq/material';
import { useGeoTools } from '../hooks';
import PropTypes from 'prop-types';
export default function GeoTools({ name, type }) {
    const { toggleProps, setLocation, ActiveTool, value } = useGeoTools(
            name,
            type
        ),
        { View } = useComponents(),
        { Toggle } = useInputComponents(),
        singleRow = useMinWidth(480);

    if (singleRow) {
        return /*#__PURE__*/ React.createElement(
            View,
            {
                style: {
                    display: 'flex',
                    flexDirection: 'row',
                    alignItems: 'center'
                }
            },
            /*#__PURE__*/ React.createElement(
                View,
                {
                    style: {
                        marginRight: 8,
                        minWidth: 200
                    }
                },
                /*#__PURE__*/ React.createElement(Toggle, { ...toggleProps })
            ),
            /*#__PURE__*/ React.createElement(ActiveTool, {
                name: name,
                value: value,
                type: type,
                setLocation: setLocation
            })
        );
    } else {
        return /*#__PURE__*/ React.createElement(
            React.Fragment,
            null,
            /*#__PURE__*/ React.createElement(
                View,
                null,
                /*#__PURE__*/ React.createElement(Toggle, {
                    label: 'Location Mode',
                    ...toggleProps
                })
            ),
            /*#__PURE__*/ React.createElement(
                View,
                {
                    style: {
                        display: 'flex',
                        flexDirection: 'row',
                        alignItems: 'center',
                        padding: 8,
                        width: '100%'
                    }
                },
                /*#__PURE__*/ React.createElement(ActiveTool, {
                    name: name,
                    value: value,
                    type: type,
                    setLocation: setLocation
                })
            )
        );
    }
}
GeoTools.propTypes = {
    name: PropTypes.string,
    type: PropTypes.string
};
