// @wq/map/components/Legend.js
import React from 'react';
import PropTypes from 'prop-types';
export default function Legend({ children }) {
    return /*#__PURE__*/ React.createElement(
        'table',
        null,
        /*#__PURE__*/ React.createElement(
            'thead',
            null,
            /*#__PURE__*/ React.createElement(
                'tr',
                null,
                /*#__PURE__*/ React.createElement('th', null, 'Group'),
                /*#__PURE__*/ React.createElement('th', null, 'Name'),
                /*#__PURE__*/ React.createElement('th', null, 'Active'),
                /*#__PURE__*/ React.createElement('th', null, 'Detail')
            )
        ),
        /*#__PURE__*/ React.createElement('tbody', null, children)
    );
}
Legend.propTypes = {
    children: PropTypes.node
};
export function BasemapToggle({ name, active, children }) {
    return /*#__PURE__*/ React.createElement(
        'tr',
        null,
        /*#__PURE__*/ React.createElement('td', null, 'Basemap'),
        /*#__PURE__*/ React.createElement('td', null, name),
        /*#__PURE__*/ React.createElement('td', null, active ? 'Y' : 'N'),
        /*#__PURE__*/ React.createElement('td', null, children)
    );
}
BasemapToggle.propTypes = {
    name: PropTypes.string,
    active: PropTypes.bool,
    children: PropTypes.node
};
export function OverlayToggle({ name, active, children }) {
    return /*#__PURE__*/ React.createElement(
        'tr',
        null,
        /*#__PURE__*/ React.createElement('td', null, 'Overlay'),
        /*#__PURE__*/ React.createElement('td', null, name),
        /*#__PURE__*/ React.createElement('td', null, active ? 'Y' : 'N'),
        /*#__PURE__*/ React.createElement('td', null, children)
    );
}
OverlayToggle.propTypes = {
    name: PropTypes.string,
    active: PropTypes.bool,
    children: PropTypes.node
};
