// @wq/react/components/ImagePreview.js
import React, { useState } from 'react';
import { useComponents } from '../hooks';
import PropTypes from 'prop-types';
export default function ImagePreview({ value }) {
    const [open, setOpen] = useState(false),
        { View, Popup, Image } = useComponents();

    if (!value) {
        return null;
    }

    const label = value.split('/').reverse()[0];
    return /*#__PURE__*/ React.createElement(
        React.Fragment,
        null,
        /*#__PURE__*/ React.createElement(Image, {
            src: value,
            alt: label,
            style: {
                maxHeight: 200,
                cursor: 'pointer'
            },
            onClick: () => setOpen(true)
        }),
        /*#__PURE__*/ React.createElement(
            Popup,
            {
                open: open,
                onClose: () => setOpen(false)
            },
            /*#__PURE__*/ React.createElement(
                View,
                {
                    style: {
                        display: 'flex',
                        justifyContent: 'center',
                        padding: 8
                    }
                },
                /*#__PURE__*/ React.createElement(Image, {
                    src: value,
                    alt: label,
                    style: {
                        maxWidth: '95vw',
                        maxHeight: '90vh'
                    }
                })
            )
        )
    );
}
ImagePreview.propTypes = {
    value: PropTypes.string
};
