// @wq/react/views/DefaultDetail.js
import React from 'react';
import {
    useRenderContext,
    useRouteInfo,
    useComponents,
    useReverse
} from '../hooks';
export default function DefaultDetail() {
    const reverse = useReverse(),
        context = useRenderContext(),
        { page, item_id, page_config } = useRouteInfo(),
        { ScrollView, PropertyTable, Fab } = useComponents(),
        form = page_config.form || [
            {
                name: 'label'
            }
        ],
        editUrl = reverse(`${page}_edit`, item_id);
    return /*#__PURE__*/ React.createElement(
        React.Fragment,
        null,
        /*#__PURE__*/ React.createElement(
            ScrollView,
            null,
            /*#__PURE__*/ React.createElement(PropertyTable, {
                form: form,
                values: context
            })
        ),
        page_config.can_change !== false &&
            /*#__PURE__*/ React.createElement(Fab, {
                icon: 'edit',
                to: editUrl
            })
    );
}
