// @wq/react/views/Login.js
import React from 'react';
import {
    useComponents,
    useInputComponents,
    useReverse,
    useRenderContext
} from '../hooks';
export default function Login() {
    const reverse = useReverse(),
        {
            Message,
            Form,
            FormError,
            HorizontalView,
            CancelButton,
            SubmitButton,
            View,
            Text,
            ButtonLink
        } = useComponents(),
        { Input } = useInputComponents(),
        { is_authenticated, user } = useRenderContext();

    if (is_authenticated) {
        return /*#__PURE__*/ React.createElement(
            View,
            null,
            /*#__PURE__*/ React.createElement(
                Text,
                null,
                'Logged in as ',
                user.label || user.username
            ),
            /*#__PURE__*/ React.createElement(
                HorizontalView,
                null,
                /*#__PURE__*/ React.createElement(
                    ButtonLink,
                    {
                        to: reverse('logout')
                    },
                    'Log Out'
                ),
                /*#__PURE__*/ React.createElement(
                    ButtonLink,
                    {
                        to: reverse('index')
                    },
                    'Return to Home'
                )
            )
        );
    }

    return /*#__PURE__*/ React.createElement(
        React.Fragment,
        null,
        /*#__PURE__*/ React.createElement(
            Form,
            {
                action: 'login',
                method: 'POST',
                storage: 'temporary',
                backgroundSync: false
            },
            /*#__PURE__*/ React.createElement(Input, {
                name: 'username',
                label: 'Username'
            }),
            /*#__PURE__*/ React.createElement(Input, {
                name: 'password',
                type: 'password',
                label: 'Password'
            }),
            /*#__PURE__*/ React.createElement(FormError, null),
            /*#__PURE__*/ React.createElement(
                HorizontalView,
                null,
                /*#__PURE__*/ React.createElement(
                    CancelButton,
                    {
                        to: reverse('index')
                    },
                    /*#__PURE__*/ React.createElement(Message, {
                        id: 'CANCEL'
                    })
                ),
                /*#__PURE__*/ React.createElement(
                    SubmitButton,
                    null,
                    /*#__PURE__*/ React.createElement(Message, {
                        id: 'SUBMIT'
                    })
                )
            )
        )
    );
}
