// @wq/material/inputs/Radio.js
import React from 'react';
import { Field } from 'formik';
import { RadioGroup } from 'formik-material-ui';
import FormControl from '@material-ui/core/FormControl';
import FormLabel from '@material-ui/core/FormLabel';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import MuiRadio from '@material-ui/core/Radio';
import HelperText from './HelperText';
import PropTypes from 'prop-types';
export default function Radio({ choices, label, ...rest }) {
    return /*#__PURE__*/ React.createElement(
        FormControl,
        {
            component: 'fieldset',
            fullWidth: true,
            margin: 'dense'
        },
        /*#__PURE__*/ React.createElement(
            FormLabel,
            {
                component: 'legend'
            },
            label
        ),
        /*#__PURE__*/ React.createElement(
            Field,
            {
                component: RadioGroup,
                ...rest
            },
            choices.map(({ name, label }) =>
                /*#__PURE__*/ React.createElement(FormControlLabel, {
                    key: name,
                    value: name,
                    label: label,
                    control: /*#__PURE__*/ React.createElement(MuiRadio, null)
                })
            )
        ),
        /*#__PURE__*/ React.createElement(HelperText, {
            name: rest.name,
            hint: rest.hint
        })
    );
}
Radio.propTypes = {
    choices: PropTypes.arrayOf(PropTypes.object),
    label: PropTypes.string
};
