// @wq/material/components/Chip.native.js
import React from 'react';
import { useIcon } from '@wq/react';
import { Chip as PaperChip, useTheme } from 'react-native-paper';
import PropTypes from 'prop-types';
export default function Chip({
    label,
    icon,
    color,
    onClick,
    onPress,
    onDelete,
    onClose,
    ...rest
}) {
    const theme = useTheme(),
        Icon = useIcon(icon);

    if (color === 'primary') {
        color = theme.colors.primary;
    } else if (color === 'secondary') {
        color = theme.colors.accent;
    }

    let style, chipTheme;

    if (color) {
        style = {
            backgroundColor: color
        };
        chipTheme = {
            colors: {
                text: '#ffffff'
            }
        };
    }

    if (!onPress) {
        onPress = onClick;
    }

    if (!onClose) {
        onClose = onDelete;
    }

    return /*#__PURE__*/ React.createElement(
        PaperChip,
        {
            icon: Icon,
            style: style,
            theme: chipTheme,
            onPress: onPress,
            onClose: onClose,
            ...rest
        },
        label
    );
}
Chip.propTypes = {
    label: PropTypes.node,
    icon: PropTypes.string,
    color: PropTypes.string,
    onClick: PropTypes.func,
    onPress: PropTypes.func,
    onDelete: PropTypes.func,
    onClose: PropTypes.func
};
