// @wq/material/components/ExpansionPanel.js
import React from 'react';
import MuiExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import PropTypes from 'prop-types';
export default function ExpansionPanel({ summary, children, open, onToggle }) {
    let handleToggle;

    if (onToggle) {
        handleToggle = (evt, state) => onToggle(state);
    }

    return /*#__PURE__*/ React.createElement(
        MuiExpansionPanel,
        {
            expanded: open,
            onChange: handleToggle
        },
        /*#__PURE__*/ React.createElement(
            ExpansionPanelSummary,
            {
                expandIcon: /*#__PURE__*/ React.createElement(
                    ExpandMoreIcon,
                    null
                )
            },
            summary
        ),
        /*#__PURE__*/ React.createElement(
            ExpansionPanelDetails,
            null,
            /*#__PURE__*/ React.createElement(
                'div',
                {
                    style: {
                        flex: 1
                    }
                },
                children
            )
        )
    );
}
ExpansionPanel.propTypes = {
    summary: PropTypes.node,
    children: PropTypes.node,
    open: PropTypes.bool,
    onToggle: PropTypes.func
};
