// @wq/material/components/ListItem.js
import React from 'react';
import MuiListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import PropTypes from 'prop-types';
import { useIcon } from '@wq/react';
export default function ListItem({
    children,
    description,
    icon,
    secondaryAction,
    ...rest
}) {
    const Icon = useIcon(icon);
    return /*#__PURE__*/ React.createElement(
        MuiListItem,
        { ...rest },
        Icon &&
            /*#__PURE__*/ React.createElement(
                ListItemIcon,
                null,
                /*#__PURE__*/ React.createElement(Icon, null)
            ),
        /*#__PURE__*/ React.createElement(ListItemText, {
            primary: children,
            secondary: description
        }),
        secondaryAction &&
            /*#__PURE__*/ React.createElement(
                ListItemSecondaryAction,
                null,
                secondaryAction
            )
    );
}
ListItem.propTypes = {
    children: PropTypes.node,
    description: PropTypes.node,
    icon: PropTypes.string,
    secondaryAction: PropTypes.node
};
