// @wq/react/inputs/File.js
import React from 'react';
import { useField, ErrorMessage } from 'formik';
import PropTypes from 'prop-types';
export default function File({ name, label, ...rest }) {
    const [, { value, initialValue }, { setValue }] = useField(name);

    function setFile(evt) {
        if (evt.target.files) {
            const { name, type } = evt.target.files[0];
            setValue({
                name,
                type,
                body: evt.target.files[0]
            });
        } else {
            setValue(initialValue || null);
        }
    }

    const hasValue = value && value !== '__clear__';
    return /*#__PURE__*/ React.createElement(
        'div',
        {
            style: {
                marginBottom: '0.5em'
            }
        },
        /*#__PURE__*/ React.createElement(
            'div',
            {
                style: {
                    display: 'flex'
                }
            },
            /*#__PURE__*/ React.createElement(
                'label',
                {
                    htmlFor: name,
                    style: {
                        width: '25%'
                    }
                },
                label
            ),
            /*#__PURE__*/ React.createElement('input', {
                style: {
                    flex: 1
                },
                name: name,
                onChange: setFile,
                ...rest,
                type: 'file'
            })
        ),
        hasValue &&
            /*#__PURE__*/ React.createElement(
                'p',
                null,
                'Current: ',
                value.name ? value.name : value,
                /*#__PURE__*/ React.createElement(
                    'button',
                    {
                        onClick: () =>
                            setValue(initialValue ? '__clear__' : null)
                    },
                    'Clear'
                )
            ),
        /*#__PURE__*/ React.createElement(ErrorMessage, {
            name: name
        })
    );
}
File.propTypes = {
    name: PropTypes.string,
    label: PropTypes.string
};
