// @wq/react/inputs/Radio.js
import React from 'react';
import { Field, ErrorMessage } from 'formik';
import PropTypes from 'prop-types';
export default function Radio({ name, label, choices, horizontal }) {
    const style = {
        display: 'flex',
        flexDirection: horizontal ? 'row' : 'column'
    };
    return /*#__PURE__*/ React.createElement(
        'div',
        {
            style: {
                marginBottom: '0.5em'
            }
        },
        /*#__PURE__*/ React.createElement(
            'fieldset',
            null,
            /*#__PURE__*/ React.createElement('legend', null, label),
            /*#__PURE__*/ React.createElement(
                'div',
                {
                    style: style
                },
                choices.map(({ name: value, label }) =>
                    /*#__PURE__*/ React.createElement(
                        'label',
                        {
                            key: value
                        },
                        /*#__PURE__*/ React.createElement(Field, {
                            name: name,
                            type: 'radio',
                            value: value
                        }),
                        label
                    )
                )
            )
        ),
        /*#__PURE__*/ React.createElement(ErrorMessage, {
            name: name
        })
    );
}
Radio.propTypes = {
    name: PropTypes.string,
    type: PropTypes.string,
    label: PropTypes.string,
    horizontal: PropTypes.bool,
    choices: PropTypes.arrayOf(PropTypes.object)
};
