// @wq/react/inputs/Select.js
import React from 'react';
import { Field, ErrorMessage } from 'formik';
import PropTypes from 'prop-types';
export default function Select({ name, type, label, choices }) {
    const multiple = type === 'select';
    return /*#__PURE__*/ React.createElement(
        'div',
        {
            style: {
                marginBottom: '0.5em'
            }
        },
        /*#__PURE__*/ React.createElement(
            'div',
            {
                style: {
                    display: 'flex'
                }
            },
            /*#__PURE__*/ React.createElement(
                'label',
                {
                    htmlFor: name,
                    style: {
                        width: '25%'
                    }
                },
                label
            ),
            /*#__PURE__*/ React.createElement(
                Field,
                {
                    as: 'select',
                    style: {
                        flex: 1
                    },
                    name: name,
                    multiple: multiple
                },
                !multiple &&
                    /*#__PURE__*/ React.createElement(
                        'option',
                        {
                            value: ''
                        },
                        'Select one...'
                    ),
                choices.map(({ name, label }) =>
                    /*#__PURE__*/ React.createElement(
                        'option',
                        {
                            key: name,
                            value: name
                        },
                        label
                    )
                )
            )
        ),
        /*#__PURE__*/ React.createElement(ErrorMessage, {
            name: name
        })
    );
}
Select.propTypes = {
    name: PropTypes.string,
    type: PropTypes.string,
    label: PropTypes.string,
    choices: PropTypes.arrayOf(PropTypes.object)
};
